internal/proposals/migration-hooks/migration-hooks-proposal.md

```diff
@@ -16,28 +16,41 @@ To achieve a valid solution for the PoC we need to come up with a design for the
-Furthermore, the `JobManager`package has a `duration` variable for benchmarking, and a `targetVersion` variable to know which jobs should be triggered at a certain deploy.
+To fulfill the requirements, a new package, called `JobManager`, is introduced, which registers, manages, and triggers certain jobs to have a fully-automated installation, or migration. This package has two (hash)maps to manage the workload: One for `pre`-jobs and one for `post`-jobs. In the (hash)maps, the key is the name of the component the jobs belong to, and the value is a slice of the jobs.
+
+Furthermore, the `JobManager` package has a `duration` variable for benchmarking.
 
-Jobs are implemented within the `JobManager` package in `go`-files, one for each component, using the specific `job`-interface. Then, the implemented interface is registered using `register(job)` in the same file. This function queues the jobs into a slice, because until then the value of the targetVersion is unknown. 
+Jobs are implemented within the `JobManager` package in `go`-files, one for each component, using the specific `job`-interface. Then, the implemented interface is registered using `register(job)` in the same file.
 
-The JobManager is used by the `deployment` package in the `deployment.go` and the `deletion.go` file. Within the `NewDeployment` or `NewDeletion` functions, the `SetKymaVersion` function of the JobManager is called to set the targetVersion, and thus to build the needed maps. Doing it in this way, we only have the required jobs registered in the maps and save some checks later on (&#8594; everything pre-calculated). Then, at the hooks, during the deployment or deletion phase, each hook only has to check if the key for the wanted component is present in the pre/post-map. If it's present, the jobs in the map are trigged, if not, nothing must be done.
+The JobManager is used by the `deployment` package in the `deployment.go` file. At the hooks, during the deployment phase, each hook only has to check if the key for the wanted component is present in the pre/post-map. If it's present, the jobs in the map are trigged, if not, nothing must be done.
 
 To benchmark the jobs, a timer is used in the pre- and post-job triggers.
 
-Retries for the jobs are not handled by the JobManager. Retries should be implemented by the jobs themselves, because it's more flexible and the interface is easy to manage.
+Retries for the jobs are not handled by the JobManager. Retries should be implemented by the jobs themselves, because it's more flexible and the interface is easy to manage. Also, the check if the logic of the job should be executed stays inside of the job, and is not implemented by the JobManager.
+
+### Deprecation of Jobs
+
+Passive:
+- Tag at which certain Kyma version jobs should be deprecated as a comment above the register call.
```

Comment:
Are the jobs deprecated as a comment? Or is the tag a comment? Assuming the second meaning.

```suggestion
- Tag (as a comment above the register call) at which defined Kyma version the jobs should be deprecated.
```